.github/workflows/functional-tests.yml

```diff
@@ -29,5 +29,15 @@ jobs:
         find src -name spec.yaml -print0 | xargs -0 -n1 aac validate
         find src/aac/plugins -name *.yaml -print0 | xargs -0 -n1 aac validate
 
+    - name: Validate Example Model
+      run: |
+        aac validate model/flow/System.yaml
+
+    - name: Test gen-plugin
+      run: |
+        mkdir gen_plugin_generation_test
+        cp  src/aac/spec/spec.yaml gen_plugin_generation_test
+        yes y | aac gen-plugin gen_plugin_generation_test/spec.yaml
```

Comment:
This produced a "gen-plugin" error: https://github.com/jondavid-black/AaC/runs/4540414112?check_suite_focus=true#step:8:9

We but the pipeline passed - we should make sure we exit with a failure (so that the pipeline fails) when this command fails or it could cause us problems later on. That said, I think we may be able to tackle this correctly with #147.

Reply:
I've added a new AC to that story to make sure we check that.

Reply:
We could make an issue to create a test bash script that runs the plugin and then asserts the expected files were generated.

This leads into another thought; I've had several QOL improvement thoughts that could be implemented with simple scripts, but I've been reluctant to introduce a scripts directory. I'm a lot more amicable to supporting a scripts directory if it includes functional test scripts like what this step needs.

Reply:
I'm not opposed to having/supporting a scripts directory to give us QOL improvements for developing the core. If it makes maintenance/testing easier and/or more effective, I think it could be worth it assuming they're small scripts. If the scripts in this directory get big enough that we start needing tests for them, etc then that could start to outweigh the benefit, though, so we'll have to keep an eye on that for sure.

That said, I think what I'm doing with my next story will handle this specific case of making sure the pipeline fails if any error is generated. It obviously won't cover checking files, etc but we could create the story for that, still.

Reply:
Changes to make:
1. Use the [gen protobuf plugin ](https://github.com/jondavid-black/AaC/blob/main/src/aac/plugins/gen_protobuf/gen_protobuf.yaml) yaml instead 
2. If broken pipe persists, this may be a [solution](https://stackoverflow.com/questions/20573282/hudson-yes-standard-output-broken-pipe)